### PR TITLE
Support students' view of locked and obtained achievements 

### LINKED ISSUE #617: Use Icon to Indicate Achievement Status
Depends on #624 

Show the achievement icon if user gained the achievement, otherwise show a locked icon (or grey icon ?).
screen shot 2015-11-25 at 10 33 51 am

###  Tasks

- [ ] Support icons for achievement 
- [ ] Use icons and background color to indicate the status of achievement

### PR DESCRIPTION
**Pull request by weiqingtoh**
This PR depends on #756. Basically, this implements the logic (mostly in helpers) required to support obtained and locked achievements. 
- Add logo and helper method for locked achievements 
- Add logic to display locked and obtained achievements

Once implemented, will resolve #617, and also complete another task on #624.

### CODE REVIEW

**Comment by lowjoel** [comment]
@weiqingtoh can you rebase and fix the comments? thanks!

**Comment by weiqingtoh** [comment]
@lowjoel should be done - let me know if any issues!

File: app/models/components/course/achievements_ability_component.rb [hunk 1]
```diff
@@ -22,6 +23,12 @@ def allow_staff_manage_achievements
     can :manage, Course::Achievement, course_staff_hash
   end
 
+  def allow_students_with_achievement_show_badges
+    can :display_badge, Course::Achievement do |achievement|
```
Comment (lowjoel) [inline app/models/components/course/achievements_ability_component.rb]: Use a hash condition, don't use blocks.

File: spec/features/course/achievement_listing_spec.rb [hunk 2]
```diff
@@ -7,18 +7,16 @@
...
-      login_as(user, scope: :user)
-      visit course_achievements_path(course)
-    end
+    before { login_as(user, scope: :user) }
 
     context 'As an Course Manager' do
       let(:user) { create(:course_manager, :approved, course: course).user }
-
-      scenario 'I can view achievements' do
+      before { visit course_achievements_path(course) }
```
Comment (lowjoel) [inline spec/features/course/achievement_listing_spec.rb]: do not use before blocks in feature specs

**Comment by lowjoel** [comment]
@weiqingtoh the specs do not pass on Travis.

**Comment by weiqingtoh** [comment]
Updated - conversation is here: https://github.com/coursemology-collab/coursemology2/commit/c8cdb202f873cab0947850939c2a4792fcc39105#commitcomment-16026623

**Pull request merged**